azure/services/vmssextensions/spec.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vmssextensions
+
+import (
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2021-04-01/compute"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/pkg/errors"
+)
+
+// VMSSExtensionSpec defines the specification for a VM or VMScaleSet extension.
+type VMSSExtensionSpec struct {
+	Name              string
```

Comment:
instead of duplicating fields here and in VMExtensionSpec (since they are the same) consider nesting the Azure extension spec in both

```
type VMSSExtensionSpec struct {
    azure.ExtensionSpec
}
```